aeronet_vector/aeronet_vector/featurecollection.py

```diff
@@ -5,60 +5,114 @@
 from rasterio.errors import CRSError
 from .feature import Feature
 from .utils import utm_zone, CRS_LATLON
+from typing import Callable
 
 
 class FeatureCollection:
     """A set of Features with the same CRS"""
 
     def __init__(self, features, crs=CRS.from_epsg(4326)):
-        self.crs = crs
+        self._crs = crs
         self.features = self._valid(features)
 
         # create indexed set for faster processing
         self.index = rtree.index.Index()
         for i, f in enumerate(self.features):
             self.index.add(i, f.bounds, f.shape)
 
+    @property
+    def crs(self):
+        return self._crs
+
+    @crs.setter
+    def crs(self, value):
+        # Not reprojecting, just setting new value
+        self._crs = value
+        for f in self.features:
+            f.crs = value
+
     def __getitem__(self, item):
         return self.features[item]
 
     def __len__(self):
         return len(self.features)
 
     def _valid(self, features):
+        # TODO: make it static?
         valid_features = []
         for f in features:
             if not f.geometry.get('coordinates'):  # remove possible empty shapes
-                warnings.warn('Empty geometry detected. This geometry have been removed from collection.',
+                warnings.warn('Empty geometry detected. This geometry has been removed from collection.',
                               RuntimeWarning)
             else:
                 valid_features.append(f)
         return valid_features
 
-    def apply(self, func):
-        return FeatureCollection([f.apply(func) for f in self.features], crs=self.crs)
+    def apply(self,  func: Callable, inplace: bool = True):
+        """Applies function to collection geometries
+        Args:
+            func (Callable): function to apply
+            inplace (bool): if True modifies collection inplace, else returns new collection
+        Returns:
+            new FeatureCollection if inplace, else None
+        """
+        if inplace:
+            for f in self.features:
+                f.apply(func, True)
+        else:
+            return FeatureCollection([f.apply(func) for f in self.features], crs=self.crs)
 
-    def filter(self, func):
-        return FeatureCollection(filter(func, self.features), crs=self.crs)
+    def filter(self, func: Callable, inplace: bool = True):
+        """Filters collection according to func
+        Args:
+            func (Callable): filtering function
+            inplace (bool): if True modifies collection inplace, else returns new collection
+        Returns:
+            new FeatureCollection if inplace, else None
+        """
+        if inplace:
+            self.features = list(filter(func, self.features))
+        else:
+            return FeatureCollection(filter(func, self.features), crs=self.crs)
 
-    def sort(self, key, reverse=False):
+    def sort(self, key: Callable, reverse: bool = False):
```

Comment:
You can't sort it without re-indexing, because index addresses by order in the FC.
Same applies to filter, I believe.

You should either modify index (remove feature, etc.) or rebuild it from scratch.